homeassistant/components/sensor/gtfs.py

```diff
@@ -228,71 +301,201 @@ def unit_of_measurement(self):
     @property
     def device_state_attributes(self):
         """Return the state attributes."""
-        return self._attributes
+        return {k: self._attributes[k] for k in sorted(self._attributes)}
 
     @property
     def icon(self):
         """Icon to use in the frontend, if any."""
+        self._icon = ICONS.get(self._route_type, ICON)
```

Comment:
Moved here from `update()`.

Reply:
properties should not have side effects, so keep this inside `update()`

Reply:
OK, reverted!